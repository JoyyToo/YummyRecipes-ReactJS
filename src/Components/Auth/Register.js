import React, { Component }  from 'react';
import MuiThemeProvider from 'material-ui/styles/MuiThemeProvider';
import TextField  from 'material-ui/TextField';
import { Link } from 'react-router-dom';
import Paper from 'material-ui/Paper';
import {notify} from 'react-notify-toast'
import '../styles.css'
import axiosInstance from '../Constants/Axioscall';


class Register extends Component {
    // Initialize state
    constructor (props) {
        super(props);
    this.state = {
        username: '',
        email: '',
        password: '',
    }
}

    // handle user input
    handleInputChange = (event) => {
        const target = event.target;
        const value = target.value;
        const name = target.name;
        this.setState({[name]: value});
    } 

    // handle register user request
    handleRegister = (event) => {
      const payload = new FormData()
      payload.set('email', this.state.email)    
      payload.set('username', this.state.username,)
      payload.set('password', this.state.password,)

        axiosInstance.post('auth/register', payload)

        .then((response) => {
            localStorage.setItem('username', this.state.username)
            notify.show(response.data.message, 'success', 4000);
            this.props
            .history
            .push('/login');
        })

        .catch((error) => {
            if (error.response){
                notify.show(error.response.data.message,'error', 4000);
            }
           
        });
    }

    // render register form
    render(){     
        let style = {
            marginLeft: 20,
            width: 323,
          };
        let divstyle = {
            width: 430,
            margin: 'auto',
            textAlign: 'left',
            paddingLeft: 20,
        }
        return (
            <MuiThemeProvider>
            <div className="main">
                <div className="tab-group">
                <Link to="/register">
                    <li className="tab active">Sign Up</li></Link>
                <Link to="/login">
                    <li className="tab">Log In</li></Link>
                </div>
                <div className="inputs">
                <Paper zDepth={2} style={divstyle}>
                    <p className="heading">Register Here</p>

                    <i className="material-icons">person</i>
                    <TextField floatingLabelText="Username" name="username" 
                                 value={this.state.username} style={style} onChange={this.handleInputChange}/><br />

                    <i className="material-icons">mail</i>
                    <TextField floatingLabelText="Email" name="email" 
                                value={this.state.email} style={style} onChange={this.handleInputChange}/><br />

                    <i className="material-icons">lock</i>
                    <TextField floatingLabelText="Password" type="password" name="password" 
                                 value={this.state.password} style={style} onChange={this.handleInputChange}/><br /><br /><br />

                    <div className="buttons">
                    <button className="network" id="one" onClick={(event => this.handleRegister(event))}
                    >Register</button>
                    <Link to="/"><button className="network" id="two">Cancel</button></Link>
                  
                    <p>Already have an account? <Link to="/login" type="button" id="button" >Login</Link></p>            <br />    
                    </div>

                </Paper><br />
                    
                </div>
            </div>
            </MuiThemeProvider>
        );
    }}

export default Register;