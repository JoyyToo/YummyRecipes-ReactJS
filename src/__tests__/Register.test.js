import React from 'react';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';

import Register from '../Components/Auth/Register'

describe('Register component', () => {
    const wrapper = shallow(<Register />);
    const preventDefault = jest.fn();
  
    it('renders properly', () => {
      expect(shallowToJson(wrapper)).toMatchSnapshot();
    });

    it('renders 1 <Register/> component', () =>{
        const component = shallow(<Register/>);
        expect(component).toHaveLength(1);
    });

    it('it returns props correctly', () =>{
        const component = shallow(<Register name="app" />);
        expect(component.instance().props.name).toBe('app')
        expect(component.instance().handleRegister( {preventDefault} ))    
    });

    it('has initial state', () => {
        expect(wrapper.state().username).toEqual('');
        expect(wrapper.state().email).toEqual('');
        expect(wrapper.state().password).toEqual('');
    });

    it('changes state', () => {
        wrapper.setState({ username: 'user', email: 'user@mail.com', password: 'password' });
        expect(wrapper.find('[name="username"]').props().value).toEqual('user');
        expect(wrapper.find('[name="email"]').props().value).toEqual('user@mail.com');
        expect(wrapper.find('[name="password"]').props().value).toEqual('password');
    });

    it('has the correct form fields', () => {
        expect(wrapper.find('[name="username"]')).toHaveLength(1);
        expect(wrapper.find('[name="email"]')).toHaveLength(1);
        expect(wrapper.find('[name="password"]')).toHaveLength(1);
    });

    it('contains MuiThemeProvider', () => {
        expect(wrapper.find('MuiThemeProvider')).toHaveLength(1);
    });

    it('contains divs', () => {
        expect(wrapper.find('div')).toHaveLength(4);
    });

    it('contains Link', () => {
        expect(wrapper.find('Link')).toHaveLength(4);
    });

    it('contains li', () => {
        expect(wrapper.find('li')).toHaveLength(2);
    });

    it('contains Paper', () => {
        expect(wrapper.find('Paper')).toHaveLength(1);
    });

    it('contains p', () => {
        expect(wrapper.find('p')).toHaveLength(2);
    });

    it('contains i', () => {
        expect(wrapper.find('i')).toHaveLength(3);
    });

    it('contains TextField ', () => {
        expect(wrapper.find('TextField')).toHaveLength(3);
    });

    it('contains button', () => {
        expect(wrapper.find('button')).toHaveLength(2);
    }); 

})
