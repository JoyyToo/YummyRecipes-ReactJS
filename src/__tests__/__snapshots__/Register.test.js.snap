// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Register component renders properly 1`] = `
<MuiThemeProvider>
  <div
    className="main"
  >
    <div
      className="tab-group"
    >
      <Link
        replace={false}
        to="/register"
      >
        <li
          className="tab active"
        >
          Sign Up
        </li>
      </Link>
      <Link
        replace={false}
        to="/login"
      >
        <li
          className="tab"
        >
          Log In
        </li>
      </Link>
    </div>
    <div
      className="inputs"
    >
      <Paper
        circle={false}
        rounded={true}
        style={
          Object {
            "margin": "auto",
            "paddingLeft": 20,
            "textAlign": "left",
            "width": 430,
          }
        }
        transitionEnabled={true}
        zDepth={2}
      >
        <p
          className="heading"
        >
          Register Here
        </p>
        <i
          className="material-icons"
        >
          person
        </i>
        <TextField
          disabled={false}
          floatingLabelFixed={false}
          floatingLabelText="Username"
          fullWidth={false}
          multiLine={false}
          name="username"
          onChange={[Function]}
          rows={1}
          style={
            Object {
              "marginLeft": 20,
              "width": 323,
            }
          }
          type="text"
          underlineShow={true}
          value=""
        />
        <br />
        <i
          className="material-icons"
        >
          mail
        </i>
        <TextField
          disabled={false}
          floatingLabelFixed={false}
          floatingLabelText="Email"
          fullWidth={false}
          multiLine={false}
          name="email"
          onChange={[Function]}
          rows={1}
          style={
            Object {
              "marginLeft": 20,
              "width": 323,
            }
          }
          type="text"
          underlineShow={true}
          value=""
        />
        <br />
        <i
          className="material-icons"
        >
          lock
        </i>
        <TextField
          disabled={false}
          floatingLabelFixed={false}
          floatingLabelText="Password"
          fullWidth={false}
          multiLine={false}
          name="password"
          onChange={[Function]}
          rows={1}
          style={
            Object {
              "marginLeft": 20,
              "width": 323,
            }
          }
<<<<<<< HEAD
<<<<<<< HEAD
          type="password"
=======
          type="text"
>>>>>>> [Feature 155119223]:
=======
          type="password"
>>>>>>> [chore] Add tests
          underlineShow={true}
          value=""
        />
        <br />
        <br />
        <br />
        <div
          className="buttons"
        >
          <button
            className="network"
            id="one"
            onClick={[Function]}
          >
            Register
          </button>
          <Link
            replace={false}
            to="/"
          >
            <button
              className="network"
              id="two"
            >
              Cancel
            </button>
          </Link>
          <p>
            Already have an account? 
            <Link
              id="button"
              replace={false}
              to="/login"
              type="button"
            >
              Login
            </Link>
          </p>
                      
          <br />
        </div>
      </Paper>
      <br />
    </div>
  </div>
</MuiThemeProvider>
`;
